contracts/ethereum/src/PaymentRegistry.sol

```diff
@@ -56,67 +49,55 @@ contract PaymentRegistry is Initializable, OwnableUpgradeable, UUPSUpgradeable {
+    function transfer(uint256 orderId, address destAddress, Chain chainId) external payable onlyOwnerOrMM {
+        require(destAddress != address(0), "Invalid address");
         require(msg.value > 0, "Invalid amount, should be higher than 0.");
 
         bytes32 index = keccak256(abi.encodePacked(orderId, destAddress, msg.value, chainId)); //200 gas
-        require(transfers[index].isUsed == false, "Transfer already processed."); //3000 gas
+        require(transfers[index] == false, "Transfer already processed."); //3000 gas (todo verify gas)
+        transfers[index] = true; //now this transfer is in progress
 
+        //old version: 
         // transfers[index] = TransferInfo({destAddress: destAddress, amount: msg.value, isUsed: true, chainId: chainId}); //65000 gas
-        // 64900 gas:
-        transfers[index].destAddress = destAddress;
-        transfers[index].amount = msg.value;
-        transfers[index].isUsed = true;
-        transfers[index].chainId = chainId;
-
 
-        (bool success,) = payable(address(uint160(destAddress))).call{value: msg.value}(""); //34000 gas
-        // (bool success,) = payable(marketMaker).call{value: msg.value}(""); //32000 gas //to implement this, address must be changed to from uint256 to addr
+        (bool success,) = payable(destAddress).call{value: msg.value}(""); //34000 gas
 
         require(success, "Transfer failed.");
-        emit Transfer(orderId, msg.sender, transfers[index]); //3000 gas
+        emit Transfer(orderId, msg.sender, destAddress, msg.value, chainId); //3000 gas (todo verify gas)
     }
 
-//TODO change name to claimPaymentStarknet
-    function claimPayment(uint256 orderId, uint256 destAddress, uint256 amount) external payable onlyOwnerOrMM {
+    function claimPayment(uint256 orderId, address destAddress, uint256 amount) external payable onlyOwnerOrMM {
         bytes32 index = keccak256(abi.encodePacked(orderId, destAddress, amount, Chain.Starknet));
-        TransferInfo storage transferInfo = transfers[index];
-        require(transferInfo.isUsed == true, "Transfer not found.");
-
-        uint256[] memory payload = new uint256[](5); //TODO why array of 256 if then filled with 128?
-        payload[0] = uint128(orderId); // low
-        payload[1] = uint128(orderId >> 128); // high
-        payload[2] = transferInfo.destAddress;
-        payload[3] = uint128(amount); // low
-        payload[4] = uint128(amount >> 128); // high
-            
+        require(transfers[index] == true, "Transfer not found.");
+
+        //changed now only uint256[3];
+        uint256[] memory payload = new uint256[](3);
+        payload[0] = uint256(orderId);
+        payload[1] = uint256(uint160(destAddress));
+        payload[2] = uint256(amount);
+        
         _snMessaging.sendMessageToL2{value: msg.value}(
             StarknetEscrowAddress,
             StarknetEscrowClaimPaymentSelector,
             payload);
-
```

Comment:
Why did u remove the emit?

Reply:
No one is currently reading this event, so I removed to save on Gas.

Do you think it is a necessary event? Who would need to read it?

Reply:
It think it is necessary to track the correct execution of the methods

Reply:
It was decided the Emit must remain for debugging, i have added it back